### PR TITLE
Fixes edge cases when checking if collider is in field of view

### PR DESCRIPTION
**Pull request by julenka**
## Overview
Fixes a few edge cases in `CameraFovChecker.IsInFOVCached` when dealing with large colliders and add tests to verify the fixes. Specifically, fix the following cases:

1. Large collider where one corner of collider is in FOV but outside the fov cone. Fix by checking for entire FOV instead of just an FOV cone.
image

2. Large collider where all corners are outside of the FOV but in front. Variations of cases like below
image

## Changes
- Fixes: #7258 
- Change method CameraFovChecker.IsInFOV*Cone*Cached to CameraFovChecker.IsInFOVCached because using cone does not cover case 1, above.
- Add test to ensure this does not regress.

## Verification
* Verify that new test does not pass without this fix
* Test HandInteractionExample scene in editor
* Validate that performance doesn't significantly regress (see below)

### Performance
Ran the following check similar to #6870 

Code instrumented in SpherePointer.cs only, around the following call:

```
if (ignoreCollidersNotInFOV)
{
    Profiler.BeginSample("IsInFOV");
    if (!mainCam.IsInFOVCached(collider))
    {
        // Additional check: is grabbable in the camera frustrum
        // We do this so that if grabbable is not visible it is not accidentally grabbed
        // Also to not turn off the hand ray if hand is near a grabbable that's not actually visible
        grabbable = null;
    }
    Profiler.EndSample();
}
```

WORST CASE 64 grabbables, none visible, 2 hands up and near the grabbables
* total time: ~0.4ms
* calls: 27

GOOD CASE 64 grabbables, all visible, 2 hands up
* total  time: ~0.2ms
* calls: 4

### CODE REVIEW

File: Assets/MixedRealityToolkit/Utilities/CameraFOVChecker.cs [hunk 1]
```diff
@@ -46,36 +47,52 @@ public static class CameraFOVChecker
                 inFOVConeLastCalculatedFrame = Time.frameCount;
             }
             else if (inFOVConeColliderCache.TryGetValue(cameraColliderPair, out result))
-            {
+            {
```
Comment (wiwei) [inline Assets/MixedRealityToolkit/Utilities/CameraFOVChecker.cs]:
```suggestion
            {
```

File: Assets/MixedRealityToolkit/Utilities/CameraFOVChecker.cs [hunk 2]
```diff
@@ -22,22 +22,23 @@ public static class CameraFOVChecker
...
-            if(myCollider.bounds.size == Vector3.zero || myCollider.transform.localScale == Vector3.zero)
+            if (myCollider.bounds.size == Vector3.zero || myCollider.transform.localScale == Vector3.zero)
             {
                 return false;
             }
-            
+
             Tuple<Collider, Camera> cameraColliderPair = new Tuple<Collider, Camera>(myCollider, cam);
             bool result = false;
             if (inFOVConeLastCalculatedFrame != Time.frameCount)
```
Comment (wiwei) [inline Assets/MixedRealityToolkit/Utilities/CameraFOVChecker.cs]: Nit: There are still some members named with "cone" - it might be worth clearing that name as well now that it's a non-cone bounds check (i.e. IsInFOVCached instead of Cone cached)

File: Assets/MixedRealityToolkit.Tests/PlayModeTests/PointerTests.cs [hunk 3]
```diff
@@ -101,15 +101,96 @@ public IEnumerator TestLinePointers()
...
+
+            yield return TestPointerFieldOfViewLargeColliderHelper(spherePointer, cube, rightHand);
+            yield return TestPointerFieldOfViewLargeColliderHelper(pokePointer, cube, rightHand);
+
+            rightHand.Hide();
+            GameObject.Destroy(cube);
+        }
+
+
+        private IEnumerator TestPointerFieldOfViewLargeColliderHelper(IMixedRealityPointer myPointer, GameObject cube, TestHand testHand)
```
Comment (Troy-Ferrell) [inline Assets/MixedRealityToolkit.Tests/PlayModeTests/PointerTests.cs]: nit: very minor but maybe TestPointerFOVLargeColliderHelper to simplify a bit

File: Assets/MixedRealityToolkit.Tests/PlayModeTests/PointerTests.cs [hunk 4]
```diff
@@ -101,15 +101,96 @@ public IEnumerator TestLinePointers()
...
+                foreach (var yOffset in yOffsets)
+                {
+                    foreach (var xOffset in xOffsets)
+                    {
+                        var cameraPos = CameraCache.Main.transform.position;
+                        var pos = new Vector3(cameraPos.x + xOffset, cameraPos.y + yOffset, cameraPos.z + zOffset);
+                        cube.transform.position = pos;
+                        yield return testHand.MoveTo(cube.transform.position);
+                        yield return PlayModeTestUtilities.WaitForInputSystemUpdate();
+                        bool isInFov = CameraCache.Main.IsInFOVCached(cube.GetComponent<BoxCollider>());
```
Comment (Troy-Ferrell) [inline Assets/MixedRealityToolkit.Tests/PlayModeTests/PointerTests.cs]: nit: can getComponent of the box collider be pulled out of the for loops?

File: Assets/MixedRealityToolkit.Tests/PlayModeTests/PointerTests.cs (hunk 3 above)
Comment (Troy-Ferrell) [reply Assets/MixedRealityToolkit.Tests/PlayModeTests/PointerTests.cs]: Also nit: can this be placed at bottom of class as private?

**Comment by davidkline-ms** [comment]
Nice fix!

**Comment by davidkline-ms** [comment]
@juleka, feel free to merge when you are ready!

File: Assets/MixedRealityToolkit/Utilities/CameraFOVChecker.cs (hunk 2 above)
Comment (julenka) [reply Assets/MixedRealityToolkit/Utilities/CameraFOVChecker.cs]: Fixed.

File: Assets/MixedRealityToolkit.Tests/PlayModeTests/PointerTests.cs (hunk 3 above)
Comment (julenka) [reply Assets/MixedRealityToolkit.Tests/PlayModeTests/PointerTests.cs]: Done

File: Assets/MixedRealityToolkit.Tests/PlayModeTests/PointerTests.cs (hunk 4 above)
Comment (julenka) [reply Assets/MixedRealityToolkit.Tests/PlayModeTests/PointerTests.cs]: Done
